src/Display/Slider/Slider.tsx

```diff
@@ -10,218 +10,193 @@ import LeftArrow from './LeftArrow';
 
     if (initialItem !== undefined) {
       if (initialItem > 0 && initialItem <= childrenCount) {
-        this.setState({
-          translateValue: -(windowWidth * (initialItem - 1)),
-          index: initialItem,
-        });
+        setTranslateValue(-(windowWidth * (initialItem - 1)));
+        setIndex(initialItem);
       } else {
-        this.setState({
-          translateValue: 0,
-          index: 1,
-        });
+        setTranslateValue(0);
+        setIndex(1);
       }
     }
-
-    if (autoplay) {
-      this.interval = setInterval(() => {
-        // eslint-disable-next-line react/destructuring-assignment
-        if (this.state.index != childrenCount) {
-          this.setState(prevState => ({
-            index: prevState.index + 1,
-            translateValue: -(prevState.screenSize * prevState.index),
-          }));
-        } else {
-          this.setState(() => ({
-            index: 1,
-            translateValue: -(screenSize * index),
-          }));
-        }
-      }, 6000);
-    }
-    window.addEventListener('resize', this.setSize);
-  }
-
-  UNSAFE_componentWillReceiveProps(nextProps: Props) {
-    const windowWidth = this.getSliderContainerDOMNode().getBoundingClientRect()
-      .width;
-
-    this.setState({
-      translateValue: -(windowWidth * (nextProps.initialItem - 1)),
-      index: nextProps.initialItem,
-    });
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('resize', this.setSize);
-    clearInterval(this.interval);
-  }
-
-  render() {
-    const {
-      children,
-      className,
-      fullContent,
-      arrowWhite,
-      removeDots,
-    } = this.props;
-    const { translateValue, index } = this.state;
-    const childrenCount = React.Children.toArray(children).filter(
-      child => !isNil(child)
-    ).length;
-
-    return (
-      <SliderContainer
-        ref={this.sliderContainerRef}
-        className={classNames('aries-slider', className)}
-        onKeyDown={this.handleKeyDown}
-        fullContent={fullContent}
-        tabIndex={0}
+  }, []);
+
+  React.useEffect(
+    function registerAutoplay() {
+      const setSize = () => {
+        const windowWidth = getSliderContainerDOMNode().getBoundingClientRect()
+          .width;
+        setScreenSize(
+          getSliderContainerDOMNode().getBoundingClientRect().width
+        );
+        setTranslateValue(-(windowWidth * (index - 1)));
+      };
+
+      const childrenCount = React.Children.toArray(children).filter(
+        child => !isNil(child)
+      ).length;
+      if (autoplay) {
+        interval.current = setInterval(() => {
+          if (index != childrenCount) {
+            setIndex(prevIndex => prevIndex + 1);
+            setTranslateValue(-(screenSize * index));
+          } else {
+            setIndex(1);
+            setTranslateValue(0);
+          }
+        }, 6000);
+      }
+      window.addEventListener('resize', setSize);
+
+      return function componentWillUnmount() {
```

Comment:
this event listener is being registered and unregistered when `index` or `screenSize` changes, so maybe call this function `unregisterAutoplay`?